### PR TITLE
Fix/event edit localization

### PR DESCRIPTION
**Pull request by kycapstick**
This PR re-writes the location name autocomplete to be handled by a custom functionality and allows the location type to be store on the location post so that when it autocompletes it will update the form. It also fixes the localization error where "Online event" was always showing in English.

### CODE REVIEW

File: plugins/events-manager/forms/event/location-moz.php [hunk 1]
```diff
@@ -182,7 +231,18 @@
 					<span class="in-person"><?php esc_html_e( 'City *', 'community-portal' ); ?></span>
 					<span class="online"><?php esc_html_e( 'URL *', 'community-portal' ); ?></span>
 				</label>
-				<input class="event-creator__input" id="location-town" type="text" name="location_town" data-string="<?php esc_html_e( 'Online Event', 'community-portal' ); ?>" value="<?php echo esc_attr( $em_location->location_town ); ?>"  maxlength="180" required/>
+				<input 
+					class="event-creator__input" 
+					id="location-town" 
+					type="text" 
+					name="location_town" 
+					data-string="<?php esc_html_e( 'Online Event', 'community-portal' ); ?>"
```
Comment (dnaluz) [inline plugins/events-manager/forms/event/location-moz.php:239]: this should be esc_attr_e

File: plugins/events-manager/forms/event/location-moz.php [hunk 2]
```diff
@@ -113,40 +134,68 @@
...
 			<div class="em-location-data-address wide--full">
 				<label class="event-creator__label" for="location-address"><?php esc_html_e( 'Address *', 'community-portal' ); ?></label>
-				<input class="event-creator__input" id="location-address" type="text" name="location_address" required value="<?php $em_location->location_address ? print esc_attr( $em_location->location_address ) : esc_html_e( 'Online', 'community-portal' ); ?>" required/> 
+				<input 
+					class="event-creator__input" 
+					id="location-address" 
+					type="text" 
+					name="location_address" 
+					required 
+					value="<?php $em_location->location_address ? print esc_attr( $em_location->location_address ) : esc_html_e( 'Online', 'community-portal' ); ?>"
```
Comment (dnaluz) [inline plugins/events-manager/forms/event/location-moz.php:155]: this should be esc_attr_e

**Comment by jveens** [comment]
@ky I had some problems testing this, And I’m never sure if I’m missing days that might be required to see some of these issues.
Could we please link to issues if they are in the repo, and include steps to test if possible?

**Comment by kycapstick** [comment]
For this PR I re-wrote the location autocomplete functionality for the Event edit/creation flow. 

**FOR TESTING**
Use either Create New Group or Edit Existing group, both use the same form. 

**Test Location Type toggle:**
When you change the location type is set to online it should make the following changes:

- Change the label for the location-name input to "Online Meeting Link"
- Hide the location-address input 
- Make the value of the location-address input "online"
- Change the label of the location-country input to "Where will this event be held?" 

Screen Shot 2020-07-16 at 12 06 31 PM

When you change the location type to in-person it should make the following changes:

- Change the label for the location-name input to "Location Name"
- Display the location-address input 
- Make the value of the location-address input an empty string
- Change the label of the location-country input to "Country" 
 
**Test Location Name Autocomplete**
When you type in the location-name input it should generate a set of suggestions for exiting locations (you need to type a minimum of 3 characters)
If you click on one of those options it should populate all of the location fields:

- location-type
- location-country
- location-name
- location-address 
- location-town

If should then set all of the inputs to read-only and the selects to disabled, and set their tabindex to -1 so they're removed from the flow.

Depending on the location-type of the selected location, it should update the ui to correspond (as outlined above).

Since you are setting both of the selects (location type and location country) to disabled they have corresponding hidden fields which should be populated with their values as otherwise the information would not be passed with the $_POST object. 

It should also set the location input box into edit mode and display the Edit Location button: 

Screen Shot 2020-07-16 at 12 19 21 PM

**TESTING LOCATION EDIT BUTTON** 
If you click the Edit Location button it should: 

- Clear all values except Location Type 
- Remove readonly attribute from all inputs, the disabled attribute from all selects, and set tabindex back to 1

** MISCELLANEOUS **

- Test ability to save event with pre-filled location
- Test ability to save with new location
- Generate errors by trying to submit without filling out location details, test that using autocomplete clears these errors

**Comment by kycapstick** [comment]
Thanks @jveens! Added a fix for that and merging everything in.

**Pull request merged**